READMEs/publish-flow-onchain.md

```diff
@@ -50,4 +68,42 @@ That's it! You've created a data asset of "SmartContractCall" asset type. It inc
 
 ## 3.  Consume the dataset
 
-(Consume here is just like in [consume-flow](READMEs/consume-flow.md]. The file downloaded is a .json. From that, use the python `json` library to parse it as desired.)
+```python
+# Alice mints and send datatokens to Bob
+to_address = bob_wallet.address
+amt_tokens = ocean.to_wei(10)  # just need 1, send more for spare
+datatoken.mint(to_address, amt_tokens, alice_wallet)
+
+
+# userdata: parameters required for the contract call 
```

Comment:
From this line onwards, it's needing 40+ lines to do what really should be one line (do order + consume), or two lines (do order; then consume). Like how [consume_flow.md](https://github.com/oceanprotocol/ocean.py/blob/main/READMEs/consume-flow.md) does it for other a url with: 
```python
file_name = ocean.assets.download_file(asset.did, bob_wallet)
```

I tried using that exact line here, but it gave me an error. Traceback:
```text
>>> file_name = ocean.assets.download_file(asset.did, bob_wallet)
Resolve did...
Order access...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/trentmc/code/play/venv/lib/python3.10/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
  File "/home/trentmc/code/play/venv/lib/python3.10/site-packages/ocean_lib/ocean/ocean_assets.py", line 771, in download_file
    order_tx_id = self.pay_for_access_service(asset, wallet)
  File "/home/trentmc/code/play/venv/lib/python3.10/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
  File "/home/trentmc/code/play/venv/lib/python3.10/site-packages/ocean_lib/ocean/ocean_assets.py", line 849, in pay_for_access_service
    raise AssetNotConsumable(consumable_result)
ocean_lib.agreements.consumable.AssetNotConsumable: 2
```

What I recommend:
- get this README shipped, with the 40+ lines (but no errors)
- and create a new github issue for more compact consume. The solution will be using ocean_assets.py, either changing the `download_file()` method to fix the error, or doing a new method.

Reply:
You can not call the function that you mention because it was made for contract functions that does not receive parameters.  Not passing the `userdata` will return a "missing parameters" message error from provider, because the function can not be evaluated without them.

Reply:
Yes. I know you can't call it as-is. I just gave it as an example, to show that it can't be used out of the box. I should have been more clear.